### PR TITLE
Refactor transaction component

### PR DESCRIPTION
**Pull request by blackdevelopa**
**Description**

Abstracts the gas polling logic from the transaction component.

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Screenshots/Recordings**

_If applicable, add screenshots or recordings to visualize the changes_

**Issue**

Progresses #???

### CODE REVIEW

File: app/core/gasPolling.ts [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    conversionRate,
+    currentCurrency,
+    transactionState,
+    ticker,
+    selectedGasFee: { suggestedGasLimit, suggestedGasPrice },
+  });
+  return parsedTransationData;
+};
+
+export const useGasFeeEstimates = () => {
```
Comment (sethkfman) [inline app/core/gasPolling.ts:166]: Since these are common core components, I would add method comments to them.

File: app/core/gasPolling.ts [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+  transactionState: any;
+  contractExchangeRates: any;
+  conversionRate: number;
+  currentCurrency: string;
+  nativeCurrency: string;
+  suggestedGasLimit: string;
+}
+
+interface LegacyProps {
```
Comment (sethkfman) [inline app/core/gasPolling.ts:106]: Similar to my comment on the functions I would add comments to the interface

File: app/core/gasPolling.test.ts [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+  transaction: {
+    value: '0xde0b6b3a7640000',
+    data: undefined,
+  },
+};
+
+describe('GasPolling', () => {
+  const token = undefined;
+  const { GasFeeController }: any = Engine.context;
+  it('should call the start gas polling controller', async () => {
```
Comment (sethkfman) [inline app/core/gasPolling.test.ts:116]: Add a negative test case. (e.g. Make the GasFeeController undefined)

File: app/core/gasPolling.ts (hunk 1 above)
Comment (blackdevelopa) [reply app/core/gasPolling.ts:191]: Updated. Thank you

**Review by sethkfman — APPROVED** [review]
LGTM

**Comment by chrisleewilcox** [comment]
Ran detox automated test.  Most test pass.  Only one that fails is deeplink and contract-nickname test fail but those automation test need to be updated.

**Review by chrisleewilcox — APPROVED** [review]
LGMT.

**Pull request merged**